django/db/backends/base/operations.py

```diff
@@ -611,7 +611,12 @@ def integer_field_range(self, internal_type):
         return a tuple of the (min_value, max_value) form representing the
         range of the column type bound to the field.
         """
-        return self.integer_field_ranges[internal_type]
+        override = {
+            'AutoField': 'IntegerField',
+            'BigAutoField': 'BigIntegerField',
+            'SmallAutoField': 'SmallIntegerField',
```

Comment:
I think we should add `SmallAutoField: (-99999, 99999)` to the `DatabaseOperations.integer_field_ranges` in the Oracle back-end.

Reply:
I put in this override thing to avoid having to update `integer_field_ranges` in all of the backends, but if any custom backends override `.integer_field_range()` without calling `super()` this will break anyway...

So I'll just update `integer_field_ranges` in all of the backends and update the release notes to state that third-party backends need to update `integer_field_ranges` for the three auto fields also...

Reply:
Maybe:
```python
if internal_type in self.integer_field_ranges:
    return self.integer_field_ranges[internal_type]
return self.integer_field_ranges[override.get(internal_type)]
```

Reply:
I opted for not overriding `.integer_field_range()` in favour of this which is much less of a kludge:
```diff
diff --git a/django/db/backends/base/operations.py b/django/db/backends/base/operations.py
index d17dc8dcce..76abc6dcb2 100644
--- a/django/db/backends/base/operations.py
+++ b/django/db/backends/base/operations.py
@@ -26,6 +26,9 @@ class BaseDatabaseOperations:
         'BigIntegerField': (-9223372036854775808, 9223372036854775807),
         'PositiveSmallIntegerField': (0, 32767),
         'PositiveIntegerField': (0, 2147483647),
+        'SmallAutoField': (-32768, 32767),
+        'AutoField': (-2147483648, 2147483647),
+        'BigAutoField': (-9223372036854775808, 9223372036854775807),
     }
     set_operators = {
         'union': 'UNION',
diff --git a/django/db/backends/oracle/operations.py b/django/db/backends/oracle/operations.py
index 016d4c9a59..3cad02fa41 100644
--- a/django/db/backends/oracle/operations.py
+++ b/django/db/backends/oracle/operations.py
@@ -16,13 +16,18 @@ from .utils import BulkInsertMapper, InsertVar, Oracle_datetime
 
 
 class DatabaseOperations(BaseDatabaseOperations):
-    # Oracle uses NUMBER(11) and NUMBER(19) for integer fields.
+    # Oracle uses NUMBER(5), NUMBER(11), and NUMBER(19) for integer fields.
+    # SmallIntegerField uses NUMBER(11) instead of NUMBER(5), which is used by
+    # SmallAutoField, to preserve backward compatibility.
     integer_field_ranges = {
         'SmallIntegerField': (-99999999999, 99999999999),
         'IntegerField': (-99999999999, 99999999999),
         'BigIntegerField': (-9999999999999999999, 9999999999999999999),
         'PositiveSmallIntegerField': (0, 99999999999),
         'PositiveIntegerField': (0, 99999999999),
+        'SmallAutoField': (-99999, 99999),
+        'AutoField': (-99999999999, 99999999999),
+        'BigAutoField': (-9999999999999999999, 9999999999999999999),
     }
     set_operators = {**BaseDatabaseOperations.set_operators, 'difference': 'MINUS'}
 
--- a/docs/releases/3.0.txt
+++ b/docs/releases/3.0.txt
@@ -392,6 +396,11 @@ backends.
 * ``DatabaseOperations.return_insert_id()`` now requires an additional
   ``field`` argument with the model field.
 
+* ``DatabaseOperations.integer_field_ranges`` has been updated to add entries
+  for ``AutoField``, ``BigAutoField``, and ``SmallAutoField`` to support the
+  standard integer range validators on these field types. Third-party backends
+  may need to override the default entries.
+
 :mod:`django.contrib.admin`
 ---------------------------
 
```